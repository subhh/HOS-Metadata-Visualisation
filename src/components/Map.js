import React, { useState, useEffect, useCallback, useContext } from 'react'
import ReactMapGL from 'react-map-gl'
import { withTheme } from 'styled-components'
import { StateContext } from 'state'
import useFetchGeo from 'hooks/useFetchGeo'
import useGeoFeatures from 'hooks/useGeoFeatures'
import useMapStyle from 'hooks/useMapStyle'
import ViewportOverlay from './ViewportOverlay'
import 'mapbox-gl/dist/mapbox-gl.css'

/**
 * Map component, displays institutions, number of publications, and clusters
 */
const Map = ({
  width = 400,
  height = 400,
  tileSize = 256,
  debug,
  theme,
  zoom,
}) => {
  const { query } = useContext(StateContext)

  const [viewport, setViewport] = useState({
    latitude: 53.5237,
    longitude: 10.02,
    zoom: 9.5,
    minZoom: 9.5,
    maxZoom: 16,
  })

  // if zoom passed via props, merge into state
  useEffect(() => {
    if (zoom && zoom !== viewport.zoom) {
      setViewport({ ...viewport, zoom })
    }
  }, [zoom, viewport])

  const handleViewportChange = useCallback(viewport => {
    setViewport(viewport)
  }, [])

  const geoLocations = useFetchGeo(query)
  const geoFeatures = useGeoFeatures(geoLocations)
  const mapStyle = useMapStyle({
    tileSize,
    geoFeatures,
    theme,
  })

  return (
    <ReactMapGL
      {...viewport}
      width={width}
      height={height}
     
      mapStyle={mapStyle}
      onViewportChange={handleViewportChange}
    >
      {debug && <ViewportOverlay viewport={viewport} />}
    </ReactMapGL>
  )
}

export default withTheme(Map)
