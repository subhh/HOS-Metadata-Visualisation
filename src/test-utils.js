import React, { useReducer } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { MemoryRouter } from 'react-router-dom'
import { StateContext, DispatchContext, reducer, initialState } from './state'
import theme from './theme'

const AllTheProviders = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <ThemeProvider theme={theme}>
      <StateContext.Provider value={state}>
        <DispatchContext.Provider value={dispatch}>
          {children}
        </DispatchContext.Provider>
      </StateContext.Provider>
    </ThemeProvider>
  )
}

const WithRouter = ({ children }) => (
  <MemoryRouter>
    <AllTheProviders>{children}</AllTheProviders>
  </MemoryRouter>
)

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export const customRenderWithRouter = (ui, options) =>
  render(ui, {
    wrapper: WithRouter,
    ...options,
  })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

export const mockAsync = data => {
  jest.spyOn(global, 'fetch').mockImplementation(url => {
    return Promise.resolve({
      json: () => Promise.resolve(data),
    })
  })
}

export const restoreAsyncMock = () => {
  global.fetch.mockRestore()
}
