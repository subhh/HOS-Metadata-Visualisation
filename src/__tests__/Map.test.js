import React from 'react'
import { render } from 'test-utils'
import Map from 'components/Map'

it('should match snapshot without data', () => {
  const { container } = render(<Map />)
  expect(container.firstChild).toMatchSnapshot()
})
