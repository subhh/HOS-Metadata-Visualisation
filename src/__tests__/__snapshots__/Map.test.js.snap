// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot without data 1`] = `
<div
  data-mock="fake-map"
/>
`;
